src/System.ComponentModel.Annotations/tests/CreditCardAttributeTests.cs

```diff
@@ -11,115 +11,94 @@ public class CreditCardAttributeTests
+        [InlineData("  1 2 3 4 5 6 7 8 9 0  1 2 34 5 2    ")]
+        [InlineData("--1-2-3-4-5-6-7-8-9-0--1-2-34-5-2----")]
+        [InlineData(" - 1- -  2 3 --4 5 6 7 -8- -9- -0 - -1 -2 -3-4- --5-- 2    ")]
+        [InlineData("1234-5678-9012-3452")]
+        [InlineData("1234 5678 9012 3452")]
+        public static void Validate_ValidValue_DoesNotThrow(string value)
         {
             var attribute = new CreditCardAttribute();
-
-            AssertEx.DoesNotThrow(() => attribute.Validate(null, s_testValidationContext)); // Null is valid
-            AssertEx.DoesNotThrow(() => attribute.Validate("0000000000000000", s_testValidationContext)); // Simplest valid value
-            AssertEx.DoesNotThrow(() => attribute.Validate("1234567890123452", s_testValidationContext)); // Good checksum
-            AssertEx.DoesNotThrow(() => attribute.Validate("1234-5678-9012-3452", s_testValidationContext)); // Good checksum, with dashes
-            AssertEx.DoesNotThrow(() => attribute.Validate("1234 5678 9012 3452", s_testValidationContext)); // Good checksum, with spaces
+            attribute.Validate(value, s_testValidationContext);
         }
 
-        [Fact]
-        public static void Validate_throws_for_invalid_values()
+        [Theory]
+        [InlineData("0000000000000001")]
+        [InlineData(0)]
+        [InlineData("000%000000000001")]
+        [InlineData("1234567890123452a")]
+        [InlineData("1234567890123452\0")]
+        public static void Validate_InvalidValue_ThrowsValidationException(object value)
         {
             var attribute = new CreditCardAttribute();
-
-            Assert.Throws<ValidationException>(() => attribute.Validate("0000000000000001", s_testValidationContext)); // Bad checksum
-            Assert.Throws<ValidationException>(() => attribute.Validate(0, s_testValidationContext)); // Non-string
-            Assert.Throws<ValidationException>(() => attribute.Validate("000%000000000001", s_testValidationContext)); // Non-digit
+            Assert.Throws<ValidationException>(() => attribute.Validate(value, s_testValidationContext));
         }
 
-        [Fact]
-        public static void Validate_throws_InvalidOperationException_if_ErrorMessage_is_null()
+        [Theory]
+        [InlineData(null, null, null)]
+        [InlineData("SomeErrorMessage", "SomeErrorMessageResourceName", null)]
+        [InlineData(null, "SomeErrorMessageResourceName", null)]
+        [InlineData(null, null, typeof(ErrorMessageResources))]
+        public static void Validate_InvalidErrorMessage_ThrowsInvalidOperationException(string message, string resourceName, Type resourceType)
         {
             var attribute = new CreditCardAttribute();
-            attribute.ErrorMessage = null; // note: this overrides the default value
+            attribute.ErrorMessage = message;
+            attribute.ErrorMessageResourceName = resourceName;
+            attribute.ErrorMessageResourceType = resourceType;
```

Comment:
I'm not sure this is actually testing the same thing as was there previously.  Setting these to null isn't the same as not setting them, e.g. https://github.com/dotnet/corefx/blob/d0dc5fc099946adc1035b34a8b1f6042eddb0c75/src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationAttribute.cs#L148-L157